styx-scheduler-service/src/main/java/com/spotify/styx/docker/KubernetesDockerRunner.java

```diff
@@ -190,9 +193,31 @@ public void init() {
         .watch(new PodWatcher(Integer.parseInt(resourceVersion)));
   }
 
-  private void pollPods() {
+  void examineRunningWFISandAssociatedPods(PodList podList) {
+    final Set<WorkflowInstance> activeWorkflowInstances = stateManager.instanceStates()
+        .parallelStream()
+        .filter(instanceState -> !instanceState.runState().equals(RUNNING))
+        .map(instanceState -> instanceState.workflowInstance())
+        .collect(toSet());
+
+    final Set<WorkflowInstance> collect = podList.getItems().parallelStream()
+        .filter(pod -> pod.getMetadata().getAnnotations()
+            .containsKey(STYX_WORKFLOW_INSTANCE_ANNOTATION))
+        .map(pod -> pod.getMetadata().getAnnotations().get(STYX_WORKFLOW_INSTANCE_ANNOTATION))
+        .map(workflowInstanceKey -> WorkflowInstance.parseKey(workflowInstanceKey))
+        .collect(toSet());
+
+    activeWorkflowInstances.removeAll(collect);
+    activeWorkflowInstances.parallelStream()
+        .forEach(workflowInstance -> stateManager.receiveIgnoreClosed(
+            Event.runError(workflowInstance, "No pod associated with this instance")));
+  }
```

Comment:
This method is not that time sensitive and I think the collections that are worked on are pretty small that the `paralellStream` use is unnecessary. It might actually be more overhead to dispatch on the fork-join-pool rather than just iterating sequentially.